lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -600,37 +599,109 @@ sub duplicate {
         $res->parents->create(
             {
-                parent_job_id => $p,
+                parent_job_id => $p->id,
                 dependency    => OpenQA::Schema::Result::JobDependencies->PARALLEL,
             });
     }
     for my $p (@direct_deps_parents_chained) {
         $res->parents->create(
             {
-                parent_job_id => $p,
+                parent_job_id => $p->id,
                 dependency    => OpenQA::Schema::Result::JobDependencies->CHAINED,
             });
     }
     for my $c (@direct_deps_children_parallel) {
         $res->children->create(
             {
-                child_job_id => $c,
+                child_job_id => $c->id,
                 dependency   => OpenQA::Schema::Result::JobDependencies->PARALLEL,
             });
     }
     for my $c (@direct_deps_children_chained) {
         $res->children->create(
             {
-                child_job_id => $c,
+                child_job_id => $c->id,
                 dependency   => OpenQA::Schema::Result::JobDependencies->CHAINED,
             });
     }
 
     # when dependency network is recreated, associate assets
     $res->register_assets_from_settings;
     # we are done, mark it in jobs_map
-    $jobs_map->{$self->id} = $res->id;
-    return ($self->id => $res->id, %duplicated_ids);
+    $jobs_map->{$self->id} = $res;
+    return ($self->id => $res, %duplicated_ids);
+}
+
+=head2 auto_duplicate
+
+=over
+
+=item Arguments: HASHREF { dup_type_auto => SCALAR, retry_avbl => SCALAR }
+
+=item Return value: ID of new job
+
+=back
+
+Handle individual job restart including associated job and asset dependencies
+
+=cut
+sub auto_duplicate {
+    my ($self, $args) = @_;
+    $args //= {};
+    # set this clone was triggered by manually if it's not auto-clone
+    $args->{dup_type_auto} //= 0;
+
+    if ($args->{dup_type_auto}) {
+        if (int($self->retry_avbl // 0) > 0) {
+            $args->{retry_avbl} = int($self->retry_avbl) - 1;
+        }
+        else {
+            log_debug("Could not auto-duplicated! The job are auto-duplicated too many times. Please restart the job manually.");
+            return;
+        }
+    }
+    else {
+        if (int($self->retry_avbl // 0) > 0) {
+            $args->{retry_avbl} = int($self->retry_avbl);
+        }
+        else {
+            $args->{retry_avbl} = 1;    # set retry_avbl back to 1
+        }
+    }
+
+    my %clones = $self->duplicate($args);
+    unless (%clones) {
+        log_debug('duplication failed');
+        return;
+    }
+    my @originals = keys %clones;
+    # abort jobs restarted because of dependencies (exclude the original $args->{jobid})
+    my $rsource = $self->result_source;
+    my $jobs    = $rsource->schema->resultset("Jobs")->search(
+        {
+            id    => {'!=' => $self->id, '-in' => \@originals},
+            state => [OpenQA::Schema::Result::Jobs::EXECUTION_STATES],
+        },
+        {
+            colums => [qw/id/]});
```

Comment:
`columns`, right? Seems like we asked for unnecessary data. Thx for @AdamWill catching the typo.